src/main/java/xades4j/providers/impl/AbstractTimeStampTokenProvider.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * XAdES4j - A Java library for generation and verification of XAdES signatures.
+ * Copyright (C) 2017 Luis Goncalves.
+ * 
+ * XAdES4j is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 3 of the License, or any later version.
+ * 
+ * XAdES4j is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with XAdES4j. If not, see <http://www.gnu.org/licenses/>.
+ */
+package xades4j.providers.impl;
+
+import com.google.inject.Inject;
+import org.apache.xml.security.algorithms.MessageDigestAlgorithm;
+import org.bouncycastle.asn1.ASN1ObjectIdentifier;
+import org.bouncycastle.asn1.cmp.PKIStatus;
+import org.bouncycastle.tsp.*;
+import xades4j.UnsupportedAlgorithmException;
+import xades4j.providers.MessageDigestEngineProvider;
+import xades4j.providers.TimeStampTokenGenerationException;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of {@code TimeStampTokenProvider} that gets time-stamp tokens
+ * from a HTTP TSA. Requests are issued with {@code certReq} set to
```

Comment:
The abstract base class is a good idea, but it shouldn't be tied to HTTP. Can you update docs?